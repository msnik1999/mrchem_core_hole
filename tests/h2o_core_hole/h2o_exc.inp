# vim:syntax=sh:

world_prec = 1.0e-3                 # Overall relative precision
world_size = 6                      # Size of simulation box 2^n

MPI {
  numerically_exact = true          # Guarantee identical results in MPI
}

Molecule {
$coords
O       0.0000  0.0000  -0.1250
H      -1.4375  0.0000   1.0250
H       1.4375  0.0000   1.0250
$end
}

WaveFunction {
  method = PBE                      # Wave function method (HF or DFT)
}

OrbitalOccupancies {
  occupancies = "0 1 0"
}

SCF {
  run = true
  max_iter = 1
  localize = false                  # Use canonical of localized orbitals
  guess_type = mw                   # Type of initial guess: none, mw, gto
  deltascf_method = IMOM
}