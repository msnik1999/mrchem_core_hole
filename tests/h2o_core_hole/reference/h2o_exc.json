{
  "input": {
    "constants": {
      "N_a": 6.02214076e+23,
      "angstrom2bohrs": 1.8897261246257702,
      "boltzmann_constant": 1.380649e-23,
      "dipmom_au2debye": 2.5417464739297717,
      "e0": 8.8541878128e-12,
      "electron_g_factor": -2.00231930436256,
      "elementary_charge": 1.602176634e-19,
      "fine_structure_constant": 0.0072973525693,
      "hartree2ev": 27.211386245988,
      "hartree2kcalmol": 627.5094740630558,
      "hartree2kjmol": 2625.4996394798254,
      "hartree2simagnetizability": 78.9451185,
      "hartree2wavenumbers": 219474.6313632,
      "light_speed": 137.035999084,
      "meter2bohr": 18897261246.2577
    },
    "geom_opt": {
      "init_step_size": -0.5,
      "max_force_component": 0.005,
      "max_history_length": 10,
      "max_iter": 100,
      "minimal_step_size": 0.01,
      "run": false,
      "subspace_tolerance": 0.001,
      "use_previous_guess": false
    },
    "molecule": {
      "charge": 0,
      "coords": [
        {
          "atom": "o",
          "r_rms": 5.0580178957e-05,
          "xyz": [
            0.0,
            0.0,
            -0.125
          ]
        },
        {
          "atom": "h",
          "r_rms": 2.6569547399e-05,
          "xyz": [
            -1.4375,
            0.0,
            1.025
          ]
        },
        {
          "atom": "h",
          "r_rms": 2.6569547399e-05,
          "xyz": [
            1.4375,
            0.0,
            1.025
          ]
        }
      ],
      "multiplicity": 1
    },
    "mpi": {
      "bank_size": -1,
      "numerically_exact": true,
      "omp_threads": -1,
      "shared_memory_size": 10000
    },
    "mra": {
      "basis_order": 5,
      "basis_type": "interpolating",
      "boxes": [
        2,
        2,
        2
      ],
      "corner": [
        -1,
        -1,
        -1
      ],
      "max_scale": 20,
      "min_scale": -5
    },
    "printer": {
      "file_name": "imom",
      "print_constants": false,
      "print_level": 0,
      "print_mpi": false,
      "print_prec": 6,
      "print_width": 75
    },
    "rsp_calculations": {},
    "scf_calculation": {
      "fock_operator": {
        "coulomb_operator": {
          "poisson_prec": 0.001,
          "shared_memory": false
        },
        "exchange_operator": {
          "exchange_prec": -1.0,
          "poisson_prec": 0.001
        },
        "kinetic_operator": {
          "derivative": "abgv_55"
        },
        "nuclear_operator": {
          "nuclear_model": "point_like",
          "proj_prec": 0.001,
          "shared_memory": false,
          "smooth_prec": 0.001
        },
        "xc_operator": {
          "shared_memory": false,
          "xc_functional": {
            "cutoff": 0.0,
            "functionals": [
              {
                "coef": 1.0,
                "name": "pbe"
              }
            ],
            "spin": false
          }
        }
      },
      "initial_guess": {
        "environment": "None",
        "external_field": "None",
        "file_CUBE_a": "cube_vectors/CUBE_a_vector.json",
        "file_CUBE_b": "cube_vectors/CUBE_b_vector.json",
        "file_CUBE_p": "cube_vectors/CUBE_p_vector.json",
        "file_basis": "initial_guess/mrchem.bas",
        "file_chk": "checkpoint/phi_scf",
        "file_gto_a": "initial_guess/mrchem.moa",
        "file_gto_b": "initial_guess/mrchem.mob",
        "file_gto_p": "initial_guess/mrchem.mop",
        "file_phi_a": "initial_guess/phi_a_scf",
        "file_phi_b": "initial_guess/phi_b_scf",
        "file_phi_p": "initial_guess/phi_p_scf",
        "localize": false,
        "method": "DFT (PBE)",
        "prec": 0.001,
        "relativity": "None",
        "restricted": true,
        "screen": 12.0,
        "type": "mw",
        "zeta": 0
      },
      "occupancies": [
        {
          "occupancy": [
            1.0,
            0.0
          ],
          "orbital": 0
        }
      ],
      "properties": {
        "dipole_moment": {
          "dip-1": {
            "operator": "h_e_dip",
            "precision": 0.001,
            "r_O": [
              0.0,
              0.0,
              0.0
            ]
          }
        }
      },
      "scf_solver": {
        "checkpoint": false,
        "deltascf_method": "IMOM",
        "energy_thrs": -1.0,
        "environment": "None",
        "external_field": "None",
        "file_chk": "checkpoint/phi_scf",
        "final_prec": 0.001,
        "helmholtz_prec": -1.0,
        "kain": 5,
        "localize": false,
        "max_iter": 1,
        "method": "DFT (PBE)",
        "orbital_thrs": 0.01,
        "relativity": "None",
        "rotation": 0,
        "start_prec": 0.001
      }
    },
    "schema_name": "mrchem_input",
    "schema_version": 1
  },
  "output": {
    "properties": {
      "center_of_mass": [
        4.6220056999883845e-18,
        0.0,
        0.003683095683220021
      ],
      "charge": 0,
      "dipole_moment": {
        "dip-1": {
          "magnitude": null,
          "r_O": [
            0.0,
            0.0,
            0.0
          ],
          "vector": [
            null,
            null,
            null
          ],
          "vector_el": [
            null,
            null,
            null
          ],
          "vector_nuc": [
            null,
            null,
            null
          ]
        }
      },
      "geometry": [
        {
          "symbol": "O",
          "xyz": [
            0.0,
            0.0,
            -0.125
          ]
        },
        {
          "symbol": "H",
          "xyz": [
            -1.4375,
            0.0,
            1.025
          ]
        },
        {
          "symbol": "H",
          "xyz": [
            1.4375,
            0.0,
            1.025
          ]
        }
      ],
      "multiplicity": 1,
      "orbital_energies": {
        "energy": [
          -21.816234401270794,
          -1.5570636691189823,
          -1.0524703563988353,
          -0.9394394205400141,
          -0.8837529779258153
        ],
        "occupation": [
          1.0,
          2.0,
          2.0,
          2.0,
          2.0
        ],
        "spin": [
          "p",
          "p",
          "p",
          "p",
          "p"
        ],
        "sum_occupied": -30.681687249238088
      },
      "orbital_positions": [
        [
          -4.045306066716214e-08,
          -2.043065981585349e-15,
          -0.12449141268465279
        ],
        [
          -4.086999856098853e-06,
          -3.562910060829435e-14,
          0.22279405801919733
        ],
        [
          -1.331635395711205e-05,
          6.29055131772979e-15,
          0.18498074708112364
        ],
        [
          5.468605231663904e-06,
          5.827718664921229e-14,
          -0.2887845095035809
        ],
        [
          -2.823547575182432e-07,
          1.541170350345445e-14,
          -0.08688060713104255
        ]
      ],
      "scf_energy": {
        "E_ee": 36.593873692370096,
        "E_eext": 0.0,
        "E_el": -65.21384243989655,
        "E_en": -147.62907016768344,
        "E_kin": 52.958758931475394,
        "E_next": 0.0,
        "E_nn": 9.039235444235468,
        "E_nuc": 9.039235444235468,
        "E_tot": -56.17460699566109,
        "E_x": 0.0,
        "E_xc": -7.137404896058594,
        "Er_el": 0.0,
        "Er_nuc": 0.0,
        "Er_tot": 0.0
      }
    },
    "provenance": {
      "creator": "MRChem",
      "mpi_processes": 1,
      "nthreads": 1,
      "routine": "mrchem.x",
      "total_cores": 1,
      "version": "1.2.0-alpha"
    },
    "rsp_calculations": null,
    "scf_calculation": {
      "initial_energy": {
        "E_ee": 33.316507976740446,
        "E_eext": 0.0,
        "E_el": -64.5798543582878,
        "E_en": -138.24377607507597,
        "E_kin": 46.932258306330745,
        "E_next": 0.0,
        "E_nn": 9.039235444235468,
        "E_nuc": 9.039235444235468,
        "E_tot": -55.54061891405234,
        "E_x": 0.0,
        "E_xc": -6.58484456628301,
        "Er_el": 0.0,
        "Er_nuc": 0.0,
        "Er_tot": 0.0
      },
      "scf_solver": {
        "converged": false,
        "cycles": [
          {
            "energy_terms": {
              "E_ee": 36.593873692370096,
              "E_eext": 0.0,
              "E_el": -65.21384243989655,
              "E_en": -147.62907016768344,
              "E_kin": 52.958758931475394,
              "E_next": 0.0,
              "E_nn": 9.039235444235468,
              "E_nuc": 9.039235444235468,
              "E_tot": -56.17460699566109,
              "E_x": 0.0,
              "E_xc": -7.137404896058594,
              "Er_el": 0.0,
              "Er_nuc": 0.0,
              "Er_tot": 0.0
            },
            "energy_total": -56.17460699566109,
            "energy_update": 0.6339880816087486,
            "mo_residual": 0.21949350787984062,
            "wall_time": 10.701201667
          }
        ],
        "wall_time": 10.703043167
      },
      "success": false
    },
    "schema_name": "mrchem_output",
    "schema_version": 1,
    "success": false
  }
}