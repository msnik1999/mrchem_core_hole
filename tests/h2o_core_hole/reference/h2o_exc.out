***************************************************************************
***                                                                     ***
***                                                                     ***
***          __  __ ____   ____ _                                       ***
***         |  \/  |  _ \ / ___| |__   ___ _ __ ___                     ***
***         | |\/| | |_) | |   | '_ \ / _ \ '_ ` _ \                    ***
***         | |  | |  _ <| |___| | | |  __/ | | | | |                   ***
***         |_|  |_|_| \_\\____|_| |_|\___|_| |_| |_|                   ***
***                                                                     ***
***         VERSION            1.2.0-alpha                              ***
***                                                                     ***
***         Git branch         mom_merged                               ***
***         Git commit hash    ef9fe2457e9ead3325c3-dirty               ***
***         Git commit author  Niklas GÃ¶llmann                         ***
***         Git commit date    Fri May 10 10:46:19 2024 +0200           ***
***                                                                     ***
***         Contact: luca.frediani@uit.no                               ***
***                                                                     ***
***         Radovan Bast            Magnar Bjorgve                      ***
***         Roberto Di Remigio      Antoine Durdek                      ***
***         Luca Frediani           Gabriel Gerez                       ***
***         Stig Rune Jensen        Jonas Juselius                      ***
***         Rune Monstad            Peter Wind                          ***
***                                                                     ***
***************************************************************************

---------------------------------------------------------------------------

 MPI processes           :       (no bank)                               1
 OpenMP threads          :                                               1
 Total cores             :                                               1

---------------------------------------------------------------------------

XCFun DFT library Copyright 2009-2020 Ulf Ekstrom and contributors.
See http://dftlibs.org/xcfun/ for more information.

This is free software; see the source code for copying conditions.
There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. For details see the documentation.
Scientific users of this library should cite
U. Ekstrom, L. Visscher, R. Bast, A. J. Thorvaldsen and K. Ruud;
J.Chem.Theor.Comp. 2010, DOI: 10.1021/ct100117s

---------------------------------------------------------------------------

 MRCPP version         : 1.6.0-alpha
 Git branch            : mom
 Git commit hash       : 9ebc1da6290a65a252e6
 Git commit author     : Laura Ratcliff
 Git commit date       : Wed Jun 14 14:11:39 2023 +0200

 Linear algebra        : EIGEN v3.4.0
 Parallelization       : OpenMP (1 threads)

---------------------------------------------------------------------------



===========================================================================
                         MultiResolution Analysis
---------------------------------------------------------------------------
 polynomial order      : 5
 polynomial type       : Interpolating
---------------------------------------------------------------------------
 total boxes           : 8
 boxes                 : [          2           2           2 ]
 unit lengths          : [   32.00000    32.00000    32.00000 ]
 scaling factor        : [    1.00000     1.00000     1.00000 ]
 lower bounds          : [  -32.00000   -32.00000   -32.00000 ]
 upper bounds          : [   32.00000    32.00000    32.00000 ]
 total length          : [   64.00000    64.00000    64.00000 ]
===========================================================================



***************************************************************************
***                                                                     ***
***                        Initializing Molecule                        ***
***                                                                     ***
***************************************************************************


===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               0
 Multiplicity            :                                               1
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0       O            :        0.000000        0.000000       -0.125000
    1       H            :       -1.437500        0.000000        1.025000
    2       H            :        1.437500        0.000000        1.025000
---------------------------------------------------------------------------
 Center of mass          :        0.000000        0.000000        0.003683
===========================================================================



***************************************************************************
***                                                                     ***
***                Computing Initial Guess Wavefunction                 ***
***                                                                     ***
***************************************************************************


Warning: guess_orbitals(), line 410: Occupancies for alpha and beta orbitals are added together for restricted calculations
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial orbitals
 Method                  : Project MW molecular orbitals
 Precision               : 1.00000e-03
 Restricted              : True
 MO file                 : initial_guess/phi_p_scf
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                            Molecular Orbitals
---------------------------------------------------------------------------
 Alpha electrons         :                                           4.500
 Beta electrons          :                                           4.500
 Total electrons         :                                           9.000
---------------------------------------------------------------------------
    n     Occ Spin       :                                            Norm
---------------------------------------------------------------------------
    0    1.00    p       :                              1.000000000000e+00
    1    2.00    p       :                              9.999999964862e-01
    2    2.00    p       :                              9.999999964858e-01
    3    2.00    p       :                              9.999998959224e-01
    4    2.00    p       :                              9.999998959224e-01
===========================================================================


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial energy
 Method                  : DFT (PBE)
 Relativity              : None
 Environment             : None
 External fields         : None
 Precision               : 1.00000e-03
 Localization            : Off
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                        Molecular Energy (initial)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                 46.932258306331
 E-N energy              :            (au)               -138.243776075076
 Coulomb energy          :            (au)                 33.316507976740
 Exchange energy         :            (au)                  0.000000000000
 X-C energy              :            (au)                 -6.584844566283
 N-N energy              :            (au)                  9.039235444235
---------------------------------------------------------------------------
 Electronic energy       :            (au)                -64.579854358288
 Nuclear energy          :            (au)                  9.039235444235
---------------------------------------------------------------------------
 Total energy            :            (au)             -5.554061891405e+01
                         :      (kcal/mol)             -3.485226456389e+04
                         :        (kJ/mol)             -1.458218749353e+05
                         :            (eV)             -1.511337233612e+03
===========================================================================


===========================================================================
                        Orbital Energies (initial)
---------------------------------------------------------------------------
    n     Occ Spin       :                                         Epsilon
---------------------------------------------------------------------------
    0    1.00    p       :            (au)                -22.485660939123
    1    2.00    p       :            (au)                 -1.790518496776
    2    2.00    p       :            (au)                 -1.242440530492
    3    2.00    p       :            (au)                 -1.144928577595
    4    2.00    p       :            (au)                 -1.116013968014
---------------------------------------------------------------------------
 Sum occupied            :            (au)                -33.073464084877
===========================================================================



***************************************************************************
***                                                                     ***
***                 Computing Ground State Wavefunction                 ***
***                                                                     ***
***************************************************************************


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Optimize ground state orbitals
 Method                  : DFT (PBE)
 Relativity              : None
 Environment             : None
 External fields         : None
 Checkpointing           : Off
 Max iterations          : 1
 KAIN solver             : 5
 Localization            : Off
 Diagonalization         : First two iterations
 Start precision         : 1.00000e-03
 Final precision         : 1.00000e-03
 Helmholtz precision     : Dynamic
 Energy threshold        : Off
 Orbital threshold       : 1.00000e-02
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
 Iter           MO residual             Total energy                Update
---------------------------------------------------------------------------
    0          2.236068e+00         -55.540618914052         -5.554062e+01
    1          2.194935e-01         -56.174606995661         -6.339881e-01
---------------------------------------------------------------------------
                        SCF did NOT converge!!!
===========================================================================



***************************************************************************
***                                                                     ***
***                    Printing Molecular Properties                    ***
***                                                                     ***
***************************************************************************


===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               0
 Multiplicity            :                                               1
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0       O            :        0.000000        0.000000       -0.125000
    1       H            :       -1.437500        0.000000        1.025000
    2       H            :        1.437500        0.000000        1.025000
---------------------------------------------------------------------------
 Center of mass          :        0.000000        0.000000        0.003683
===========================================================================


===========================================================================
                         Molecular Energy (final)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                 52.958758931475
 E-N energy              :            (au)               -147.629070167683
 Coulomb energy          :            (au)                 36.593873692370
 Exchange energy         :            (au)                  0.000000000000
 X-C energy              :            (au)                 -7.137404896059
 N-N energy              :            (au)                  9.039235444235
---------------------------------------------------------------------------
 Electronic energy       :            (au)                -65.213842439897
 Nuclear energy          :            (au)                  9.039235444235
---------------------------------------------------------------------------
 Total energy            :            (au)             -5.617460699566e+01
                         :      (kcal/mol)             -3.525009809155e+04
                         :        (kJ/mol)             -1.474864104150e+05
                         :            (eV)             -1.528588928176e+03
===========================================================================


===========================================================================
                         Orbital Energies (final)
---------------------------------------------------------------------------
    n     Occ Spin       :                                         Epsilon
---------------------------------------------------------------------------
    0    1.00    p       :            (au)                -21.816234401271
    1    2.00    p       :            (au)                 -1.557063669119
    2    2.00    p       :            (au)                 -1.052470356399
    3    2.00    p       :            (au)                 -0.939439420540
    4    2.00    p       :            (au)                 -0.883752977926
---------------------------------------------------------------------------
 Sum occupied            :            (au)                -30.681687249238
===========================================================================


===========================================================================
                             Orbital Positions
---------------------------------------------------------------------------
Index                    :               x               y               z
---------------------------------------------------------------------------
    0                    :       -0.000000       -0.000000       -0.124491
    1                    :       -0.000004       -0.000000        0.222794
    2                    :       -0.000013        0.000000        0.184981
    3                    :        0.000005        0.000000       -0.288785
    4                    :       -0.000000        0.000000       -0.086881
===========================================================================


===========================================================================
                           Dipole Moment (dip-1)
---------------------------------------------------------------------------
 r_O                     :        0.000000        0.000000        0.000000
---------------------------------------------------------------------------
 Electronic vector       :             N/A             N/A             N/A
 Magnitude               :            (au)                             N/A
                         :         (Debye)                             N/A
---------------------------------------------------------------------------
 Nuclear vector          :             N/A             N/A             N/A
 Magnitude               :            (au)                             N/A
                         :         (Debye)                             N/A
---------------------------------------------------------------------------
 Total vector            :             N/A             N/A             N/A
 Magnitude               :            (au)                             N/A
                         :         (Debye)                             N/A
===========================================================================




***************************************************************************
***                                                                     ***
***                            Exiting MRChem                           ***
***                                                                     ***
***                       Wall time :  0h  0m 18s                       ***
***                                                                     ***
***************************************************************************